src/popupmenu.c

```diff
@@ -722,10 +720,19 @@ pum_redraw(void)
 
 			    if (rt != NULL)
 			    {
-				char_u		*rt_start = rt;
-				int		cells;
+				char_u	    *rt_start = rt;
+				int	    cells;
+				int	    over_cell = 0;
+				int	    truncated = FALSE;
 
 				cells = mb_string2cells(rt , -1);
+				truncated = pum_width == p_pmw
+						&& pum_width - totwidth < cells;
```

Comment:
That looks strange. I would think if `pum_width - totwidth < cells` is true, then it shouldn't be truncated?

Reply:
totwidth is the width that the previous item has used. pum_width - totwidth is the width that pum currently has available. When the available width is not enough to accommodate the current item, doesn't this mean truncation.